### ISSUE TITLE
Messaging Server and Client

### ISSUE THREAD

**Issue by seitensei** [seitensei/GVCIS457#2]
Application level functionality for this project. We want to have an server/application combination that runs on each device, with an associated GUID or UUID (which will be advertised to the network, and used to link devices). We should be able to send and receive text-based messages to and from each device to any other device that it is connected to.

**Stretch Goal: **Not just sending text, but also images and files.

Blocked on #1.
